server/src/main/java/org/opensearch/action/admin/cluster/remotestore/stats/RemoteStoreStats.java

```diff
@@ -0,0 +1,108 @@
+
+import java.io.IOException;
+
+/**
+ * Encapsulates all remote store stats
+ *
+ * @opensearch.internal
+ */
+public class RemoteStoreStats implements Writeable, ToXContentFragment {
+
+    private final RemoteRefreshSegmentTracker.Stats remoteSegmentUploadShardStats;
+
+    public RemoteStoreStats(RemoteRefreshSegmentTracker.Stats remoteSegmentUploadShardStats) {
+        this.remoteSegmentUploadShardStats = remoteSegmentUploadShardStats;
+    }
+
+    public RemoteStoreStats(StreamInput in) throws IOException {
+        remoteSegmentUploadShardStats = in.readOptionalWriteable(RemoteRefreshSegmentTracker.Stats::new);
+    }
+
+    public RemoteRefreshSegmentTracker.Stats getStats() {
+        return remoteSegmentUploadShardStats;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject()
+            .field(Fields.SHARD_ID, remoteSegmentUploadShardStats.shardId)
+
+            .field(Fields.LOCAL_REFRESH_TIMESTAMP, remoteSegmentUploadShardStats.localRefreshTimeMs)
+            .field(Fields.LOCAL_REFRESH_CUMULATIVE_COUNT, remoteSegmentUploadShardStats.localRefreshCount)
+            .field(Fields.REMOTE_REFRESH_TIMESTAMP, remoteSegmentUploadShardStats.remoteRefreshTimeMs)
+            .field(Fields.REMOTE_REFRESH_CUMULATIVE_COUNT, remoteSegmentUploadShardStats.remoteRefreshCount)
+            .field(Fields.BYTES_LAG, remoteSegmentUploadShardStats.bytesLag)
+
+            .field(Fields.REJECTION_COUNT, remoteSegmentUploadShardStats.rejectionCount)
+            .field(Fields.CONSECUTIVE_FAILURE_COUNT, remoteSegmentUploadShardStats.consecutiveFailuresCount);
+
+        builder.startObject(Fields.TOTAL_REMOTE_REFRESH);
+        builder.field(Fields.STARTED, remoteSegmentUploadShardStats.totalUploadsStarted)
+            .field(Fields.SUCCEEDED, remoteSegmentUploadShardStats.totalUploadsSucceeded)
+            .field(Fields.FAILED, remoteSegmentUploadShardStats.totalUploadsFailed);
+        builder.endObject();
+
+        builder.startObject(Fields.TOTAL_UPLOADS_IN_BYTES);
+        builder.field(Fields.STARTED, remoteSegmentUploadShardStats.uploadBytesStarted)
+            .field(Fields.SUCCEEDED, remoteSegmentUploadShardStats.uploadBytesSucceeded)
+            .field(Fields.FAILED, remoteSegmentUploadShardStats.uploadBytesFailed);
+        builder.endObject();
+        builder.startObject(Fields.REMOTE_REFRESH_SIZE_IN_BYTES);
+        builder.field(Fields.LAST_SUCCESSFUL, remoteSegmentUploadShardStats.lastSuccessfulRemoteRefreshBytes);
+        builder.field(Fields.MOVING_AVG, remoteSegmentUploadShardStats.uploadBytesMovingAverage);
+        builder.endObject();
+
+        builder.startObject(Fields.UPLOAD_LATENCY_IN_BYTES_PER_SEC);
+        builder.field(Fields.MOVING_AVG, remoteSegmentUploadShardStats.uploadBytesPerSecMovingAverage);
+        builder.endObject();
+        builder.startObject(Fields.REMOTE_REFRESH_LATENCY_IN_NANOS);
```

Comment:
Why is it in nanos? Even though user can convert it into millis/seconds, do we really need nano second level granularity?

Reply:
the value that is being set in the Stats class is actually in MS, pls rename the field accordingly.